src/main/scala/reporting/Converter.scala

```diff
@@ -0,0 +1,531 @@
+    extends ExtractedModelEntry {
+  override def toString = s"$value [$problem]"
+  def asValueEntry = ConstantEntry(value)
+}
+
+case class SeqEntry(name: String, values: List[ExtractedModelEntry])
+    extends ExtractedModelEntry {
+  override def toString = s"($name): [${values.map(_.toString).mkString(", ")}]"
+  def asValueEntry = ConstantEntry(name)
+}
+
+case class UnprocessedModelEntry(entry: ValueEntry)
+    extends ExtractedModelEntry {
+  override def toString = s"$entry"
+  def asValueEntry = entry
+}
+
+// processed Heap representation:
+sealed trait HeapEntry{
+  def toString: String
+}
+
+case class ExtractedHeap(entries: List[HeapEntry])
+
+case class PredHeapEntry(
+    name: String,
+    args: Seq[ExtractedModelEntry]
+) extends HeapEntry {
+  override def toString = s"$name(${args.mkString(", ")})"
+}
+case class FieldHeapEntry(
+    recv: VarEntry,
+    field: String,
+    perm: Option[Rational],
+    sort: Sort,
+    entry: ExtractedModelEntry
+) extends HeapEntry {
+  override def toString = s"$recv.$field"
+}
+
+case class UnresolvedHeapEntry(chunk: Chunk, reason: String) extends HeapEntry {
+  override def toString = s"$chunk (not further processed because: $reason)"
+}
+
+object Converter {
+
+  def getFunctionValue(
+      model: Model,
+      fname: String,
+      args: Seq[ValueEntry],
+      toSort: Sort
+  ): ExtractedModelEntry = {
+    val entry: Option[ModelEntry] = model.entries.get(fname)
+    entry match {
+      case Some(MapEntry(m, els)) =>
+        getConstantEntry(toSort, m.getOrElse(args, els))
+      case Some(m) => getConstantEntry(toSort, m)
+      case None    => OtherEntry("${fname}", "function not found in model")
+    }
+  }
+
+  def translateSort(s: Sort): String = {
+    s match {
+      case sorts.Set(els: Sort) => s"Set<${translateSort(els)}>"
+      case sorts.Ref            => "$Ref"
+      case sorts.Snap           => "$Snap"
+      case sorts.Perm           => "$Perm"
+      case sorts.Seq(els: Sort) => s"Seq<${translateSort(els)}>"
+      case _                    => s.toString
+    }
+  }
+
+  def getConstantEntry(s: Sort, m: ModelEntry): ExtractedModelEntry = {
+    s match {
+      case sorts.Ref => VarEntry(m.toString, sorts.Ref)
+      case sorts.Int =>
+        m match {
+          case ConstantEntry(x)             => LitIntEntry(BigInt(x))
+          case ApplicationEntry(name, args) =>
+            //this is needed because negative integers are stored as ApplicationEntries
+            val res = getConstantEntry(s, args.head)
+            (res, name) match {
+              case (LitIntEntry(x), "-") => LitIntEntry(-x)
+              case _ => OtherEntry(s"$m", "not an integer literal")
+            }
+          case _ => OtherEntry(s"$m", "not an integer literal")
+        }
+      case sorts.Bool =>
+        m.toString.toLowerCase() match {
+          case "true"  => LitBoolEntry(true)
+          case "false" => LitBoolEntry(false)
+          case x =>
```

Comment:
Well this was just me trying to play it safe, because I don't know if there are certain special cases and I would not want my code to fail because of that, ApplicationEntries of some sort like for the integers or I don't know. But if you say so I will certainly remove that part.

Reply:
Okay so while using it for prusti I actually found an example where this triggers a match error now, without the third match case.
The problem in the model is:
`$SortWrappers.$SnapToBool -> {
    (= (:var 0) $Snap.unit)
}` 
Is this even something I could resolve to a boolean somehow?

Reply:
This means ``$SortWrappers.$SnapToBool(x) = (x == $Snap.unit)``.
For what it's worth, the ModelParser in Silver used to have special support for parsing this at least in some scenarios (see function boolFuncDef here: https://github.com/viperproject/silver/blob/4dbefcad60b98b80fbffd80e87a6355c79f07ae2/src/main/scala/viper/silver/verifier/ModelParser.scala) but this isn't in the current version any more for some reason.

Reply:
Aah okay, thank you very much!
Then I guess I'll just re-implement this for my use if this is not used anywhere else.

Reply:
I'm not sure if that's what you meant, but IMO this should be implemented in the ModelParser in Silver again; the exisiting code in Nagini and 2Vyper for example also assume that it'ts there (and I think it was just removed by accident).

Reply:
Oh no then it's not what I meant.
I guess I can do it but I've not worked on silver at all so far.

Reply:
I'll give it a try. Do you happen to have an example where Z3 generates this output so that I can use it for testing?

Reply:
Yes sure.
[https://github.com/cedihegi/silicon/blob/counterexample-interface/src/test/resources/demo-counterexamples/prusti-bug.vpr](url)

Reply:
Thanks! Try this Silver branch: https://github.com/viperproject/silver/tree/marcoeilers_ce_parsing_improvement and let me know if something doesn't work.
I'll need to test it properly before I merge it into the main branch.